packages/appium/lib/utils.js

```diff
@@ -209,6 +211,50 @@ function getPackageVersion(pkgName) {
   return pkgInfo.version;
 }
 
+/**
+ * Adjusts NODE_PATH environment variable,
+ * so drivers and plugins could load their peer dependencies.
+ * Read https://nodejs.org/api/modules.html#loading-from-the-global-folders
+ * for more details.
+ * @returns {Promise<void>}
+ */
+async function adjustNodePath() {
```

Comment:
I mentioned this earlier, but I don't think this actually does anything.

Modifying `process.env.NODE_PATH` in process _A_ does not change the effective `NODE_PATH` of process _A_.  It _can_ affect the `NODE_PATH` of child process _B_.

If we do this in `init()`, only child processes will be effected.  Since extensions are _not_ loaded in child processes, it does _not_ change how extensions resolve `appium`.

As mentioned on the `npm` issue tracker, `npm` does not use `NODE_PATH` to determine what is to be installed.  So even if this change was intended to affect `npm`'s behavior, it won't.

Reply:
We can show this does not work _in vivo_ with:

Create `foo.js`:

```js
process.env.NODE_PATH='./baz';
require('bar');
```

Create dir `baz/`. Create `baz/bar.js`:

```js
module.exports = true
```

Now, run `node foo.js`, which will fail.  Run `NODE_PATH='./baz' node foo.js`, which will succeed.

Reply:
I have found a cool article about resolving dependencies and decided to use https://gist.github.com/branneman/8048520#7-the-hack as the simplest solution for now. Although, it would probably make more sense for us to use the documented solution https://gist.github.com/branneman/8048520#8-the-wrapper

Reply:
Yes, can you please use the supported solution instead?  Would rather not be touching undocumented internals...

Reply:
I really want to see some sort of E2E test

Reply:
> Yes, can you please use the supported solution instead? Would rather not be touching undocumented internals...

In order to use the documented version we would have to modify imports of appium components in all depending modules. Basically instead of `import { fs, util } from 'appium/support';` it would look like 

```js
// Somewhere in Appium codebase
import { createRequire } from 'node:module';
export const requireAppium = createRequire(<path_to_appium_root>);




// Somewhere in client modules
const { fs, util } = requireAppium('./support');
```

Reply:
https://nodejs.org/api/module.html#modulecreaterequirefilename

Reply:
yeah i really don't want to use a nonstandard require all over the place